upcase_booklist_app/upcase_booklist_app.py

```diff
@@ -1,8 +1,44 @@
-from flask import Flask 
+from flask import Flask, render_template, url_for, redirect
 app = Flask(__name__)
 
 
+
+BOOKS = {
```

Comment:
Nice data structure! You _could_ use integers, like `1`, as the keys for each book. That's how it will be in the database.